### PR TITLE
⚠️ Updated function name  HasPausedAnnotation to HasPaused and HasSkipRemediationAnnotation to HasSkipRemediation

### LINKED ISSUE #5539: rename `annotations.HasPausedAnnotation(obj)` - function name stutters
**Detailed Description**

The function name `annotations.HasPausedAnnotation` stutters. Consider renaming it to something like `annotations.HasPaused`?

**Anything else you would like to add:**

Since this will be an exposed API change we cannot directly rename the function. 
Steps to follow:
* Create the a new function `HasPaused` in the `util/annotations` package.
* Add a deprecation comment on `HasPausedAnnotation`
  * Example: `// Deprecated: use HasPaused instead.`
* Update all usages of `HasPausedAnnotation` to use `HasPaused`.

/kind feature
/help
/good-first-issue

### PR DESCRIPTION
**Pull request by tamboliasir1**
Thank you for opportunity and guidance.
Fixes #5539

### CODE REVIEW

File: util/annotations/helpers.go [hunk 1]
```diff
@@ -29,19 +29,25 @@ func IsPaused(cluster *clusterv1.Cluster, o metav1.Object) bool {
...
-	return HasPausedAnnotation(o)
+	return HasPaused(o)
 }
 
 // IsExternallyManaged returns true if the object has the `managed-by` annotation.
 func IsExternallyManaged(o metav1.Object) bool {
 	return hasAnnotation(o, clusterv1.ManagedByAnnotation)
 }
 
+//Deprecated: Please use `HasPaused` instead
```
Comment (enxebre) [inline util/annotations/helpers.go:40]: Nit: This is missing space after // and final period.

```
// Deprecated: this is going to be removed in a next release. Please use `HasPaused` instead.
```

Comment (tamboliasir1) [reply util/annotations/helpers.go:40]: Got it..

**Review by killianmuldoon — COMMENTED** [review]
Would it be a good idea to replace 'HasSkipRemediationAnnotation' in this PR too? It has the same stuttering problem.

**Comment by tamboliasir1** [comment]
Yeah sure I can do it.. Should I rename it to `HasSkipRemediation` with same deprecation message?

**Comment by killianmuldoon** [comment]
@tamboliasir1 That sounds good - it's only got one usage as far as I can see, so please update there too.

**Review by CecileRobertMichon — COMMENTED** [review]
/ok-to-test

**Comment by CecileRobertMichon** [comment]
@tamboliasir1 when you get a chance, please squash commits 1 & 2 since they are related to the same change

**Comment by fabriziopandini** [comment]
lgtm pending squash

File: util/annotations/helpers.go [hunk 2]
```diff
@@ -29,24 +29,36 @@ func IsPaused(cluster *clusterv1.Cluster, o metav1.Object) bool {
...
+	return HasPaused(o)
 }
 
 // IsExternallyManaged returns true if the object has the `managed-by` annotation.
 func IsExternallyManaged(o metav1.Object) bool {
 	return hasAnnotation(o, clusterv1.ManagedByAnnotation)
 }
 
+// Deprecated: this is going to be removed in a next release. Please use `HasPaused` instead.
 // HasPausedAnnotation returns true if the object has the `paused` annotation.
```
Comment (ykakarap) [inline util/annotations/helpers.go:41]:
```suggestion
// HasPausedAnnotation returns true if the object has the `paused` annotation.
// Deprecated: this is going to be removed in a next release. Please use `HasPaused` instead.
```

File: util/annotations/helpers.go [hunk 3]
```diff
@@ -29,24 +29,36 @@ func IsPaused(cluster *clusterv1.Cluster, o metav1.Object) bool {
...
 	return hasAnnotation(o, clusterv1.PausedAnnotation)
 }
 
+// HasPaused returns true if the object has the `paused` annotation.
+func HasPaused(o metav1.Object) bool {
+	return hasAnnotation(o, clusterv1.PausedAnnotation)
+}
+
+// Deprecated: this is going to be removed in a next release. Please use `HasSkipRemediation` instead.
 // HasSkipRemediationAnnotation returns true if the object has the `skip-remediation` annotation.
```
Comment (ykakarap) [inline util/annotations/helpers.go:52]:
```suggestion
// HasSkipRemediationAnnotation returns true if the object has the `skip-remediation` annotation.
// Deprecated: this is going to be removed in a next release. Please use `HasSkipRemediation` instead.
```

**Review by ykakarap — COMMENTED** [review]
lgtm pending squash

File: util/annotations/helpers.go (hunk 3 above)
Comment (ykakarap) [reply util/annotations/helpers.go:52]: same as above.

File: util/annotations/helpers.go (hunk 2 above)
Comment (tamboliasir1) [reply util/annotations/helpers.go:41]: Got it..

**Comment by tamboliasir1** [comment]
Hi If I updated changes for go lint now so should I squash all commits in one or should I do squash for particular commit?

**Comment by killianmuldoon** [comment]
@tamboliasir1 you can squash this into a single commit - the :seedling marking seems right to me. This isn't breaking because the old methods are just being deprecated, not removed.

**Comment by ykakarap** [comment]
@tamboliasir1 Linting still seems to be failing. 
FYI, you can run `make lint` locally to see what the problem. Generally the warning is associated with an explanation of what failed and how you can fix it.

**Comment by tamboliasir1** [comment]
Updated lint and squashed git commits.

**Comment by ykakarap** [comment]
@tamboliasir1 The PR looks good. Thanks for working on this. :) 

/lgtm

**Review by fabriziopandini — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**